storage/innobase/handler/ha_innodb.cc

```diff
@@ -1896,10 +1897,8 @@ ulonglong ha_innobase::table_version() const
   return m_prebuilt->trx_id;
 }
 
-#ifdef UNIV_DEBUG
 /** whether the DDL log recovery has been completed */
-static bool ddl_recovery_done;
-#endif
+std::atomic_bool ddl_recovery_done { false };
```

Comment:
This would open up possibilities for many race conditions. I do not think that it is a good idea.

Based on other changes around this, I understood that some binlog recovery would be executed before `innodb_ddl_recovery_done()` is invoked. It would seem cleaner to me to complete any lower-level recovery first.